modules/signalfx/test/java/com/spotify/ffwd/signalfx/SignalFxPluginSinkTest.java

```diff
@@ -0,0 +1,105 @@
+package com.spotify.ffwd.signalfx;
+
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.signalfx.metrics.flush.AggregateMetricSender;
+import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.DataPoint;
+import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.Dimension;
+import com.spotify.ffwd.model.Metric;
+import eu.toolchain.async.AsyncFramework;
+import eu.toolchain.async.TinyAsync;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SignalFxPluginSinkTest {
+
+  private SignalFxPluginSink sink;
+
+  private AsyncFramework async;
+  private Supplier<AggregateMetricSender> senderSupplier;
+
+  private final int threadCount = Runtime.getRuntime().availableProcessors();
+
+  private final ExecutorService executor = Executors.newFixedThreadPool(threadCount);
+
+  @Mock
+  private AggregateMetricSender sender;
+
+  private StubSession session = new StubSession();
+
+  private final String METRIC_KEY_NOT_EXCEEDING = "signalfx_test";
+  private final String METRIC_KEY_EXCEEDING = "signalfx_test_8217398173982139821739872198372198372198379djsfjjksdfhkdsjhfkjdsh2187392187392183798213982173982139219382198329379218739821739821723982179871298379128739281739827398273747874657678478384787498983598984759837598439583498734985793487";
+  private final String WHAT = "error-reply-ratio";
+  private final String TAG_VAL_NOT_EXCEEDING = "awsus";
+  private final String TAG_VAL_EXCEEDING = "awsuskjdfkjsfvlsdjldskfjsdlkfdskfds;kfmdkslfmsdkfjdlsjflksvmjnskjdkjsdkdbkdbv dklmlskkmxvmkmvdkmvkclx;d;skf;smvkvmdskmvdsvmdsvxmv kxmvlxmldksdmlksdmvdlsjcxzm.xmldkm;dvsmv;dskmv;dslvmlx,xmlkvxmd;md;mdknvldnvldnsvlndlsnlnvlcnvlkndlvndlvnjslnvc";
+
+  private static final int CHAR_LIMIT = 256;
+
+  @Before
+  public void setup() {
+    sink = new SignalFxPluginSink();
+    async = TinyAsync.builder().executor(executor).build();
+    senderSupplier = () -> sender;
+    sink.async = async;
+    sink.senderSupplier = senderSupplier;
+  }
+
+  @Test
+  public void sendMetricNotExceedCharLim(){
+    Metric metric = new Metric(METRIC_KEY_NOT_EXCEEDING, 1278, new Date(), "", ImmutableSet.of(), ImmutableMap.of("what", WHAT, "pod", TAG_VAL_NOT_EXCEEDING), "test_proc");
+    final List<Metric> metricsList = Collections.singletonList(metric);
+    when(sender.createSession()).thenReturn(session);
+    sink.start();
+    Future<Void> future = sink.sendMetrics(metricsList);
+    while (!future.isDone()){
```

Comment:
`future.get()`

Reply:
why?

Reply:
it blocks until the future is done. replaces the while loop.